rita_common/src/usage_tracker/mod.rs

```diff
@@ -133,44 +160,120 @@ impl UsageTracker {
-                                blank_usage_tracker
+        if Path::new(&settings::get_rita_common().network.usage_tracker_file).exists()
+            && Path::new(&settings::get_rita_common().network.usage_tracker_file)
+                .extension()
+                .unwrap()
+                .eq("bincode")
+        {
+            let file = File::open(settings::get_rita_common().network.usage_tracker_file);
+
+            match file {
+                Ok(mut file) => {
+                    let mut byte_contents = Vec::new();
+                    // try compressed
+                    match file.read_to_end(&mut byte_contents) {
+                        Ok(_bytes_read) => {
+                            let mut decoder = ZlibDecoder::new(&byte_contents[..]);
+                            // Extract data from decoder, note this is streaming decoding, so we're
+                            // decompressing the data as we take it out of the zlib compressed object
+                            trace!("attempting to unzip or read bw history");
+                            let deserialized: Result<UsageTracker, Error> =
+                                bincode::deserialize_from(&mut decoder);
+
+                            match deserialized {
+                                Ok(value) => value,
+                                Err(e) => {
+                                    error!(
+                                        "Failed to deserialize bytes in compressed bw history {:?}",
+                                        e
+                                    );
+                                    blank_usage_tracker
+                                }
                             }
                         }
+                        Err(e) => {
+                            error!("Failed to read usage tracker file! {:?}", e);
+                            blank_usage_tracker
+                        }
                     }
-                    Err(e) => {
-                        error!("Failed to read usage tracker file! {:?}", e);
-                        blank_usage_tracker
-                    }
+                }
+                Err(e) => {
+                    error!("Failed to open usage tracker file! {:?}", e);
+                    blank_usage_tracker
                 }
             }
-            Err(e) => {
-                error!("Failed to open usage tracker file! {:?}", e);
-                blank_usage_tracker
+        } else {
+            // if we get here, the bincode file does not exist yet
+            let file = File::open(settings::get_rita_common().network.usage_tracker_file);
+
+            match file {
+                Ok(mut file) => {
+                    let mut byte_contents = Vec::new();
+                    // try compressed
+                    match file.read_to_end(&mut byte_contents) {
+                        Ok(_bytes_read) => {
+                            let decoder = ZlibDecoder::new(&byte_contents[..]);
```

Comment:
can we de-duplicate this with the try compressed one on line 174, it might have to be broken out into it's own function.

Reply:
sort of fixed... not broken out into its own function, but nested match statements have been broken up. Trying to wrap serdeError and bincode Error into ritacommonerror was being a pain